test/hotspot/jtreg/testlibrary_tests/ir_framework/examples/IRExample.java

```diff
@@ -34,63 +34,95 @@
+ *                               often each IR node/user-defined regex should be matched on the compilation output of
+ *                               each compile phase.</li>
  * </ul>
  * <p>
  *
  * One might also want to restrict the application of certain @IR rules depending on the used flags in the test VM.
- * These could be flags defined by the user or by JTreg. In the latter case, the flags must be whitelisted (see
- * {@link TestFramework}) most of them should not have an impact on the IR except for different GC flags which should
- * be considered) to enable a verification by the framework (see below). The @IR rules thus have an option to restrict
- * their application:
+ * These could be flags defined by the user or by JTreg. In the latter case, the flags must be whitelisted in
+ * {@link TestFramework#JTREG_WHITELIST_FLAGS} (i.e. have no unexpected impact on the IR except if the flag simulates a
+ * specific machine setup like UseAVX={1,2,3} etc.) to enable an IR verification by the framework.
+ * The @IR rules thus have an option to restrict their application:
  * <ul>
- *     <li><p>{@link IR#applyIf}: Only apply a rule if a flag has a certain value</li>
- *     <li><p>{@link IR#applyIfNot}: Only apply a rule if a flag has NOT a certain value (inverse of applyIf)</li>
- *     <li><p>{@link IR#applyIfAnd}: Only apply a rule if all flags have the specified value</li>
- *     <li><p>{@link IR#applyIfOr}: Only apply a rule if at least one flag has the specified value</li>
+ *     <li><p>{@link IR#applyIf}:    Only apply a rule if a flag has the specified value/range of values.</li>
+ *     <li><p>{@link IR#applyIfNot}: Only apply a rule if a flag has NOT a specified value/range of values
+ *                                   (inverse of applyIf).</li>
+ *     <li><p>{@link IR#applyIfAnd}: Only apply a rule if ALL flags have the specified value/range of values.</li>
+ *     <li><p>{@link IR#applyIfOr}:  Only apply a rule if AT LEAST ONE flag has the specified value/range of values.</li>
  * </ul>
  * <p>
  *
- * The framework, however, does not perform the verification if:
+ * An IR verification cannot always be performed. Certain VM flags explicitly disable IR verification, change the IR
+ * shape in unexpected ways letting IR rules fail or even make IR verification impossible:
  * <ul>
  *     <li><p>-DVerifyIR=false is used</li>
  *     <li><p>The test is run with a non-debug build</li>
  *     <li><p>-Xcomp, -Xint, -XX:-UseCompile, -XX:CompileThreshold, -DFlipC1C2=true, or -DExcludeRandom=true are used.</li>
- *     <li><p>JTreg specifies non-whitelisted flags as VM and/or Javaoptions (could change the IR in an unexpected way)</li>
+ *     <li><p>JTreg specifies non-whitelisted flags as VM and/or Javaoptions (could change the IR in unexpected ways).</li>
  * </ul>
  *
  * @see IR
  * @see Test
  * @see TestFramework
  */
```

Comment:
You're right. I originally only had the text in the `IRExample.java` file but then I thought it might be better to have it in the `README` file as well - but then we can indeed get rid of the text in `IRExample.java`. I've cleaned that up.